File: python/unidad_02/problema_2_05.py
"""
El programa, dado como dato un número entero, determina si el
mismo es positivo, negativo o nulo.
"""

def main():

    try:

        numero = int(input("Número entero: "))

    except ValueError as e:

        print(e)
        return
    
    if numero < 0:

        print(f"{numero} es un número negativo")
        return
    
    if numero > 0:

        print(f"{numero} es un número positivo")
        return
    
    print(f"{numero} es un número natural dependiendo")

    print("del contexto o corriente de pensamiento")

if __name__ == '__main__':

    main()

File: java/unidad_02/Problema_2_05/src/Problema_2_05.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * El programa, dado como dato un número entero, determina si el
 * mismo es positivo, negativo o nulo.
 */

public class Problema_2_05 {

    public static void main(String[] args) {

        int numero;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.print("Número entero: ");
            numero = scanner.nextInt();

        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        if (numero < 0) {

            System.out.printf("%d es un número negativo", numero);
            return;
        }
        if (numero > 0) {

            System.out.printf("%d es un número positivo", numero);
            return;
        }
        System.out.printf("%d es un número natural dependiendo %n", numero);

        System.out.println("del contexto o corriente de pensamiento");
    }
}